client/lib/patch-db.ts

```diff
@@ -1,26 +1,46 @@
-import { merge, Observable, of } from 'rxjs'
-import { concatMap, finalize, tap } from 'rxjs/operators'
-import { Source } from './source/source'
+import { merge, Observable, Subject, Subscription } from 'rxjs'
 import { Store } from './store'
 import { DBCache, Http } from './types'
+import { RPCError } from './source/ws-source'
+import { Source } from './source/source'
 
 export class PatchDB<T> {
-  store: Store<T>
+  public store: Store<T> = new Store(this.http, this.initialCache)
+  public connectionError$ = new Subject<Error>()
+  public rpcError$ = new Subject<RPCError>()
+  public cache$ = new Subject<DBCache<T>>()
+
+  private updatesSub?: Subscription
+  private sourcesSub = this.sources$.subscribe(sources => {
+    this.updatesSub = merge(...sources.map(s => s.watch$(this.store))).subscribe({
+      next: (res) => {
+        console.log('PatchDB -> subscribedChanges -> JCWM:', { res })
+        if ('result' in res) {
+          this.store.update(res.result)
+          this.cache$.next(this.store.cache)
+        }
+        else {
+          this.rpcError$.next(res)
+        }
+      },
+      error: (e) => {
+        this.connectionError$.next(e)
```

Comment:
So if the connection error happens it will unsubscribe from the other sources, just pointing this out.